frontend/controller/backend.js

```diff
@@ -14,14 +14,40 @@ const languageFileMap = new Map([
 ])
 
 sbp('okTurtles.events/on', NOTIFICATION_TYPE.VERSION_INFO, (versionInfo) => {
+  if (versionInfo.GI_VERSION === process.env.GI_VERSION) {
+    // No refresh necessary, we're already at the latest version
+    sessionStorage.removeItem(NOTIFICATION_TYPE.VERSION_INFO)
+    return
+  }
+
   console.info('New Group Income version available:', versionInfo)
-  // Prevent the client from trying to reconnect when the page starts unloading.
-  sbp('okTurtles.data/get', PUBSUB_INSTANCE).destroy()
   // TODO: allow the user to manually reload the page later.
+  try {
+    // Store the current VERSION_INFO in session storage to prevent infinite
+    // reload loops
+    const existingSerialized = sessionStorage.getItem(NOTIFICATION_TYPE.VERSION_INFO)
+    if (existingSerialized) {
+      const existingVersionInfo = JSON.parse(existingSerialized)
+      if (
+        Array.isArray(existingVersionInfo) &&
+        !(Date.now() - existingVersionInfo[0] >= 1e7) &&
+        versionInfo.GI_VERSION === existingVersionInfo[1].GI_VERSION
+      ) {
+        console.warn('[NOTIFICATION_TYPE.VERSION_INFO] A different Group Income version is available, but reloading has failed to address it', { existingVersionInfo, versionInfo })
```

Comment:
Should probably be an `error`

Reply:
It's not necessarily an error condition.